### PR TITLE
Swipe rewriting

### LINKED ISSUE #20: [Feature] Cache getRecs results when navigating to other pages
At this moment when you go from swipes to somewhere else and then returning to the swipe page, another swipe card is displayed. In other words, the swipe process is re-initiated.
I believe this is one of the causes that eventually locks up the swiping and resulting in the 'Out of people' messages while there actually are enough swipes left. And even if it's not, it's still not so nice.

Suggestion: code to keep swipe page running in background.

### LINKED ISSUE #100: Last swipe is lost when navigating to another page
Each pass/like action on the swipe page is held in a queue until either the next swipe action, or 10 seconds passes, at which point the API call will be made. When you navigate to another page though, this queue is not flushed, and the API call for your last swipe is never made.

We should flush the queue when navigating to another page (and, if possible, when closing the app).

### LINKED ISSUE #101: Superlike actions are not added to the swipe queue
Superlikes currently aren't added to the API action queue on the swipe page, so they can't be undone.

### LINKED ISSUE #103: move swing to ng-swing
The code who handle the cards is 1 years old and it's use the npm swing package but now there is a very good angular directive to do that : https://github.com/gajus/angular-swing , also the code is a little messy and can be shorted using this lib, in the same time it can solve #101 and #100.

### PR DESCRIPTION
**Pull request by run1t**
Changelog
- Change swing version to 3.0.3
- Closes #103 , fixes #101 , fixes #100 and fixes #20
- preload swipe card when you have 3 left
- cache the discovery settings

Bug : 
Sometimes the pictures blink...

### CODE REVIEW

File: desktop-app/js/tinder-desktop.swipe.js [hunk 1]
```diff
@@ -14,315 +106,165 @@
...
-      };
-      var callNow = immediate && !timeout;
-      clearTimeout(timeout);
-      timeout = setTimeout(later, wait);
-      if (callNow) func.apply(context, args);
+  
+  
+  module.filter('bdayToAge', function() {
+    return function(bday) {
+      return moment.duration(moment().diff(moment(bday))).years();
```
Comment (stuwil) [inline desktop-app/js/tinder-desktop.swipe.js]: This is in tinder-desktop.common.js. Looks like the conflict was incorrectly resolved.

File: desktop-app/js/tinder-desktop.swipe.js [hunk 2]
```diff
@@ -14,315 +106,165 @@
...
+  });
 
-  module.directive('renderImagesDirective', function() {
-    return function(scope, element, attrs) {
-      if (scope.$last){
-        scope.$emit('cardsRendered');
-      }
+  module.filter('pingToAgo', function() {
+    return function(ping) {
+      return moment(ping).fromNow();
```
Comment (stuwil) [inline desktop-app/js/tinder-desktop.swipe.js]: This is in tinder-desktop.common.js. Looks like the conflict was incorrectly resolved.

File: desktop-app/js/tinder-desktop.discovery.js [hunk 3]
```diff
@@ -1,74 +1,83 @@
...
     };
     
+    /* Passport */
     $scope.autocompleteOptions = {
       types: '(cities)'
     };
     
-    $scope.watchAutocomplete = function () { return $scope.details; };
-    
+    $scope.watchAutocomplete = function () { return $scope.details; }
```
Comment (stuwil) [inline desktop-app/js/tinder-desktop.discovery.js]: Missing trailing semicolon.

**Comment by stuwil** [comment]
Thanks @run1t. Should changing discovery settings or Location force new recommendations / a new swipe stack?

**Comment by run1t** [comment]
Good point, I will add this

File: desktop-app/js/tinder-desktop.common.js [hunk 4]
```diff
@@ -17,6 +17,12 @@
     };
   });
 
+  module.filter('pingToAgo', function() {
+    return function(ping) {
+      return moment(ping).fromNow();
+    };
+  });
+
```
Comment (stuwil) [inline desktop-app/js/tinder-desktop.common.js]: Same as the timeFromNow filter. Not in use, remove please.

**Comment by run1t** [comment]
@stuwil  done ! ;)

**Comment by stuwil** [comment]
@run1t: Hrm, seems to be pulling new recommendations even when nothing is changed in Discovery?

Go to Discovery, don't change anything, and go back to the Swipe page...

**Pull request closed without merging**